File: maximum-count-of-positive-integer-and-negative-integer.ts
{
    const maximumCount = function(nums) {
        const result = {
            pos: 0,
            neg: 0,
        }

        for(const num of nums) {
            if (num > 0) {
                result.pos += 1
            }
            if (num < 0) {
                result.neg += 1
            }
        }

        return Math.max(result.pos,result.neg)
     };

    console.log(maximumCount([-2,-1,-1,1,2,3]))
}

File: maximum-count-of-positive-integer-and-negative-integer.rs
use std::cmp;

pub fn maximum_count(nums: Vec<i32>) -> i32 {
    let (mut pos, mut neg) = (0, 0);

    for num in nums {
        if num > 0 {
            pos += 1;
        } else if num < 0 {
            neg += 1
        };
    }

    pos.max(neg)
}
